app/assets/javascripts/lib/bip.js

```diff
@@ -1,780 +0,0 @@
-    this.element.trigger(jQuery.Event("ajax:error"), request, error);
-
-    // Binding back after being clicked
-    jQuery(this.activator).bind('click', {editor: this}, this.clickHandler);
-    this.element.trigger(jQuery.Event("best_in_place:deactivate"));
-  },
-
-  clickHandler : function(event) {
-    event.preventDefault();
-    event.data.editor.activate();
-  },
-
-  setHtmlAttributes : function() {
-    var formField = this.element.find(this.formType);
-
-    if(this.html_attrs){
-      var attrs = jQuery.parseJSON(this.html_attrs);
-      for(var key in attrs){
-        formField.attr(key, attrs[key]);
-      }
-    }
-  }
-};
-
-
-// Button cases:
-// If no buttons, then blur saves, ESC cancels
-// If just Cancel button, then blur saves, ESC or clicking Cancel cancels (careful of blur event!)
-// If just OK button, then clicking OK saves (careful of blur event!), ESC or blur cancels
-// If both buttons, then clicking OK saves, ESC or clicking Cancel or blur cancels
-BestInPlaceEditor.forms = {
-  "input" : {
-    activateForm : function() {
-      var output = jQuery(document.createElement('form'))
-                   .addClass('form_in_place')
-                   .attr('action', 'javascript:void(0);')
-                   .attr('style', 'display:inline');
-      var input_elt = jQuery(document.createElement('input'))
-                      .attr('type', 'text')
-                      .attr('name', this.attributeName)
-                      .val(this.display_value);
-      if(this.inner_class !== null) {
-        input_elt.addClass(this.inner_class);
-      }
-      output.append(input_elt);
-      if(this.okButton) {
-        output.append(
-          jQuery(document.createElement('input'))
-          .attr('type', 'submit')
-          .attr('class', this.okButtonClass)
-          .attr('value', this.okButton)
-        )
-      }
-      if(this.cancelButton) {
-        output.append(
-          jQuery(document.createElement('input'))
-          .attr('type', 'button')
-          .attr('class', this.cancelButtonClass)
-          .attr('value', this.cancelButton)
-        )
-      }
-
-      this.element.html(output);
-      this.setHtmlAttributes();
-      // START METAMAPS CODE
-      //this.element.find("input[type='text']")[0].select();
-      this.element.find("input[type='text']")[0].focus();
-      // END METAMAPS CODE
-      this.element.find("form").bind('submit', {editor: this}, BestInPlaceEditor.forms.input.submitHandler);
-      if (this.cancelButton) {
-        this.element.find("input[type='button']").bind('click', {editor: this}, BestInPlaceEditor.forms.input.cancelButtonHandler);
-      }
-      this.element.find("input[type='text']").bind('blur', {editor: this}, BestInPlaceEditor.forms.input.inputBlurHandler);
-      // START METAMAPS CODE
-      this.element.find("input[type='text']").bind('keydown', {editor: this}, BestInPlaceEditor.forms.input.keydownHandler);  
-      // END METAMAPS CODE
```

Comment:
i think this is also for enter key handling to tell the truth. so if i can make the enter key work, i think we're ok to ignore these hacks

unless you have more historical context on the hacks

did you write the hacks or did i?